server/segment_rpc.go

```diff
@@ -46,14 +45,15 @@ var errCapCompat = errors.New("incompatible capabilities")
 
 var tlsConfig = &tls.Config{InsecureSkipVerify: true}
 var httpClient = &http.Client{
-	Transport: &http2.Transport{
+	Transport: &http.Transport{
 		TLSClientConfig: tlsConfig,
-		DialTLS: func(network, addr string, cfg *tls.Config) (gonet.Conn, error) {
+		DialTLSContext: func(ctx context.Context, network, addr string) (gonet.Conn, error) {
```

Comment:
Any way to use this context here? Otherwise we may not respect the timeout during dial

Reply:
Fixed in 5079f86

Now we set a dial timeout, but also allow the caller of `DialTLSContext` (i.e. the `http.Transport` internally) to cancel the context if it needs to by passing the context param to a `tlsDialer.DialContext()` call.

Reply:
Great addition on an explicit dial timeout as well!